### ISSUE TITLE
[clang-tidy] False negative readability-redundant-casting with typedef and using

### ISSUE THREAD

**Issue by chrchr-github** [llvm/llvm-project#81641]
~~~c++
typedef long LONG;
using ULONG = unsigned long;
void f(long x) {
    LONG y = static_cast<LONG>(x);
}
void g(unsigned long x) {
    ULONG y = static_cast<ULONG>(x);
}
void h(long x) {
    long y = static_cast<long>(x);
}
~~~
~~~
<source>:10:14: warning: redundant explicit casting to the same type 'long' as the sub-expression, remove this casting [readability-redundant-casting]
   10 |     long y = static_cast<long>(x);
      |              ^~~~~~~~~~~~~~~~~~ ~
<source>:9:13: note: source type originates from referencing this parameter]
    9 | void h(long x) {
      |        ~~~~ ^
~~~
There are no warnings for `f()` and `g()`.
https://godbolt.org/z/cxEs3WrdK

**Comment by PiotrZSL** [comment]
Set `readability-redundant-casting.IgnoreTypeAliases : true` to detect also those cases.
Default behavior for this check is an "safe" one.

**Comment by chrchr-github** [comment]
I see, should have read the documentation first.
I still don't get the warnings here, but that might be a different issue:
https://godbolt.org/z/Gzbjr8K69

**Comment by PiotrZSL** [comment]
In last one issue is with config, you using old config format, and passing a string, proper format is:
```--checks=readability-redundant-casting --config="{Checks: 'readability-redundant-casting', CheckOptions: {readability-redundant-casting.IgnoreTypeAliases: true}}"```
Probably there is some issue with config.
